x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/EnrichFeatureSetUsage.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.enrich;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.XPackFeatureSet;
+import org.elasticsearch.xpack.core.XPackField;
+import org.elasticsearch.xpack.core.enrich.action.EnrichStatsAction.Response.CoordinatorStats;
+import org.elasticsearch.xpack.core.enrich.action.EnrichStatsAction.Response.ExecutionStats;
+
+public class EnrichFeatureSetUsage extends XPackFeatureSet.Usage {
+
+    private ExecutionStats executionStats;
+    private List<CoordinatorStats> coordinatorStats;
```

Comment:
Perhaps instead of just returning `CoordinatorStats` from all ingest nodes, we should just return min/max/avg based on the values inside these `CoordinatorStats` instances?

Reply:
This seems reasonable. I removed the -1 values from the min stats as well, since a value of 0 is easier to reason about and doesn't require any special encoding.